client/http.go

```diff
@@ -66,67 +66,86 @@ func NewHTTPClient(endpoint, token, accountID, orgID, projectID, pipelineID, bui
+	if fileExists(certFile) && fileExists(keyFile) {
+		mtlsCerts, err := tls.LoadX509KeyPair(certFile, keyFile)
 		if err != nil {
-			fmt.Printf("could not read directory %s, error: %s\n", additionalCertsDir, err)
-			client.Client = clientWithRootCAs(skipverify, rootCAs)
-			return client
+			fmt.Printf("failed to load mTLS cert/key pair, error: %s\n", err)
+			return false, tls.Certificate{}
 		}
+		return true, mtlsCerts
+	}
+	return false, tls.Certificate{}
+}
 
-		// Go through all certs in this directory and add them to the global certs
-		for _, f := range files {
-			path := filepath.Join(additionalCertsDir, f.Name())
-			fmt.Printf("trying to add certs at: %s to root certs\n", path)
-			// Create TLS config using cert PEM
-			rootPem, err := os.ReadFile(path)
-			if err != nil {
-				fmt.Printf("could not read certificate file (%s), error: %s\n", path, err.Error())
-				continue
-			}
-			// Append certs to the global certs
-			ok := rootCAs.AppendCertsFromPEM(rootPem)
-			if !ok {
-				fmt.Printf("error adding cert (%s) to pool, please check format of the certs provided.\n", path)
-				continue
-			}
-			fmt.Printf("successfully added cert at: %s to root certs\n", path)
+// loadRootCAs loads custom root CAs from the provided directory
+func loadRootCAs(additionalCertsDir string) *x509.CertPool {
+	if additionalCertsDir == "" {
+		return nil
+	}
+
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	fmt.Printf("additional certs dir to allow: %s\n", additionalCertsDir)
+
+	files, err := os.ReadDir(additionalCertsDir)
+	if err != nil {
+		fmt.Printf("could not read directory %s, error: %s\n", additionalCertsDir, err)
+		return rootCAs
+	}
+
+	// Go through all certs in this directory and add them to the global certs
+	for _, f := range files {
+		path := filepath.Join(additionalCertsDir, f.Name())
+		fmt.Printf("trying to add certs at: %s to root certs\n", path)
+		// Create TLS config using cert PEM
+		rootPem, err := os.ReadFile(path)
+		if err != nil {
+			fmt.Printf("could not read certificate file (%s), error: %s\n", path, err.Error())
+			continue
 		}
-		client.Client = clientWithRootCAs(skipverify, rootCAs)
+		// Append certs to the global certs
+		ok := rootCAs.AppendCertsFromPEM(rootPem)
+		if !ok {
+			fmt.Printf("error adding cert (%s) to pool, please check format of the certs provided.\n", path)
+			continue
+		}
+		fmt.Printf("successfully added cert at: %s to root certs\n", path)
 	}
-	return client
+	return rootCAs
 }
 
-func clientWithRootCAs(skipverify bool, rootCAs *x509.CertPool) *http.Client {
-	// Create the HTTP Client with certs
+// clientWithTLSConfig creates an HTTP client with the provided TLS settings
+func clientWithTLSConfig(skipverify bool, rootCAs *x509.CertPool, mtlsEnabled bool, cert tls.Certificate) *http.Client {
 	config := &tls.Config{
 		InsecureSkipVerify: skipverify,
 	}
 	if rootCAs != nil {
```

Comment:
if skipverify is true, shouldn't we just ignore the rootCAs and certs?

Reply:
if skipverify is true then the client does not verify the server and hence we can ignore rootCAs because they wont be used by the client to verify the server

We cannot ignore client mtls certs since the server still needs to verify the client regardless

Reply:
Modified code to not add custom rootCAs if skipverify is set to true. This will match the original logic